compiler/src/dmd/initsem.d

```diff
@@ -871,14 +884,7 @@ extern(C++) Initializer initializerSemantic(Initializer init, Scope* sc, ref Typ
                         /* Disambiguate between an exp representing the entire
                          * struct, and an exp representing the first field of the struct
                          */
-                        if (needInterpret)
-                            sc = sc.startCTFE();
-                        ExpInitializer ei = ix.isExpInitializer();
-                        ei.exp = ei.exp.expressionSemantic(sc);
-                        ei.exp = resolveProperties(sc, ei.exp);
-                        if (needInterpret)
-                            sc = sc.endCTFE();
-                        if (ei.exp.implicitConvTo(tn))      // initializer represents the entire struct
+                        if (representsStruct(ix.isExpInitializer(), tns)) // initializer represents the entire struct
```

Comment:
Why not use a UCFS call?

Reply:
Ahh, because it's a nested function.

Reply:
It's a good idea, but I'm not sure it makes much difference here.

Reply:
Yes. Not so important.